2026 reduced basis use hdf5 by default for saveload and allow basis loading in sequential

lberti opened the pull request:
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes?
- [X] Have you successfully run the mor testsuite with your changes locally?
- [ ] Have you written Doxygen comments in your contribution ?

-----

lberti committed:
Allow basis loading in sequential #2026
- charge relation functionSpace for sequential online phase

lberti committed:
fix typo

lberti committed:
Add option export-solution to crbonlinerun #2026
- activates database and elements loading

lberti committed:
up command line documentation crbonlinerun #2026

lberti committed:
#2026
- change default format to HDF5
- save primal/dual basis functions to separate files

vincentchabannes commented on mor/apps/crbonlinerun.cpp:
```diff
@@ -61,7 +61,8 @@ loadPlugin( std::string const& name, std::string const& id )
     std::cout << "Loaded the plugin " << plugin->name() << std::endl;
     bool loadFiniteElementDatabase = boption(_name="crb.load-elements-database");
-
+    if(boption("export-solution"))
```
use _name=

vincentchabannes commented on mor/mor/feel/feelmor/crb.hpp:
```diff
@@ -1514,6 +1514,7 @@ class CRB : public CRBDB,
         {}
 
     virtual void saveRB();
+    virtual void savenewRBelement();
```
keep saveRB but you can add in arg if necessary

vincentchabannes commented on mor/mor/feel/feelmor/crb.hpp:
```diff
@@ -2571,6 +2572,15 @@ CRB<TruthModelType>::offline()
+        // save the rb function space
+        if ( true )
+        {
+            std::string functionSpaceFilenameBase = (boost::format("%1%_functionSpace_p%2%.json")%this->name() %this->worldComm().size()).str();
```
replace boost format by fmt

vincentchabannes commented on mor/mor/feel/feelmor/crb.hpp:
```diff
@@ -11617,7 +11641,7 @@ CRB<TruthModelType>::saveJson()
             }
         }
-
+        std::cout << "to write json" << std::endl;
```
remove or use LOG(INFO)

vincentchabannes commented on mor/mor/feel/feelmor/crbdb.cpp:
```diff
@@ -102,13 +102,15 @@ CRBDB::lookForDB() const
 {
     //std::cout << "db localpath " << this->dbLocalPath() <<" db filename=" << this->dbFilename() << std::endl;
     // look in local repository $HOME/feel/db/crb/...
-    if ( fs::exists( this->dbLocalPath() / this->dbFilename() ) )
```
maybe be to revert

lberti committed:
#2026
- fix boption with _name
- replace boost::format with fmt::format

lberti committed:
#2026
- revert lookForDB() function to its initial state
- delete saveNewRBelement and use save() function

lberti committed:
Merge branch 'develop' into 2026-reduced-basis-use-hdf5-by-default-for-saveload-and-allow-basis-loading-in-sequential

lberti commented:
@vincentchabannes, the checks have passed. The last changes might have to be reviewed and, if everything is ok, this PR could be merged

vincentchabannes committed:
Merge branch 'develop' into 2026-reduced-basis-use-hdf5-by-default-for-saveload-and-allow-basis-loading-in-sequential
